test/integration/Sandwich.t.sol

```diff
@@ -0,0 +1,221 @@
+        _bystanderDepositAmount = bound(
+            _bystanderDepositAmount,
+            1e18,
+            hyperdrive.calculateMaxLong() / 3
+        );
+        uint256 bystanderShares = depositEverlong(
+            _bystanderDepositAmount,
+            bystander
+        );
+
+        // The attacker opens a large short.
+        _shortAmount = bound(
+            _shortAmount,
+            1e18,
+            hyperdrive.calculateMaxShort() / 2
+        );
+        (uint256 maturityTime, uint256 attackerShortBasePaid) = openShort(
+            attacker,
+            _shortAmount
+        );
+
+        // The attacker deposits into Everlong.
+        _attackerDepositAmount = bound(
+            _attackerDepositAmount,
+            1e18,
+            hyperdrive.calculateMaxLong() / 3
+        );
+        uint256 attackerShares = depositEverlong(
+            _attackerDepositAmount,
+            attacker
+        );
+
+        // The attacker closes their short position.
+        uint256 attackerShortProceeds = closeShort(
+            attacker,
+            maturityTime,
+            _shortAmount
+        );
+
+        // The attacker redeems their Everlong shares.
+        uint256 attackerEverlongProceeds = redeemEverlong(
+            attackerShares,
+            attacker
+        );
+
+        // The bystander redeems their Everlong shares.
+        redeemEverlong(bystanderShares, bystander);
+
+        // Calculate the amount paid and the proceeds for the attacker.
+        uint256 attackerPaid = _attackerDepositAmount + attackerShortBasePaid;
+        uint256 attackerProceeds = attackerEverlongProceeds +
+            attackerShortProceeds;
+
+        // Ensure the attacker does not profit.
+        assertLt(attackerProceeds, attackerPaid);
+    }
+
+    /// @dev Tests the following scenario:
+    ///      1. Attacker adds liquidity.
+    ///      2. Bystander deposits.
+    ///      3. Attacker deposits.
+    ///      4. Attacker removes liquidity.
+    ///      5. Attacker withdraws.
+    ///      6. Bystander withdraws.
+    function sandwich_lp_instant(
+        uint256 _lpDeposit,
+        uint256 _bystanderDeposit,
+        uint256 _attackerDeposit
+    ) public {
+        // Alice is the attacker, and Bob is the bystander.
+        address attacker = alice;
+        address bystander = bob;
+
+        // The attacker adds liquidity to Hyperdrive.
+        _lpDeposit = bound(_lpDeposit, 1e18, INITIAL_CONTRIBUTION);
+        uint256 attackerLPShares = addLiquidity(attacker, _lpDeposit);
+
+        // The bystander deposits into Everlong.
+        _bystanderDeposit = bound(
+            _bystanderDeposit,
+            1e18,
+            hyperdrive.calculateMaxLong() / 3
+        );
+        uint256 bystanderEverlongShares = depositEverlong(
+            _bystanderDeposit,
+            bystander
+        );
+
+        // The attacker deposits into Everlong.
+        _attackerDeposit = bound(
```

Comment:
Now that you broke up where the deposit amounts are instantiated, shouldn't we be able to get away with always using `hyperdrive.calculateMaxLong()`? Rebalance will already have been called, so `hyperdrive.calculateMaxLong()` will take that into account at this point.

Reply:
When I do this, I still get `Insufficient Liquidity` errors.

I think this is because we still attempt to rebalance on deposit and don't handle hyperdrive errors well yet...

Imo it would be most efficient time-wise to tackle this class of issues (like the one above with minTxAmount) all together, so going to tack on to that TODO and revisit later.

Wdyt?